_posts/2018-02-16-Java-Time2.md

```diff
@@ -0,0 +1,1564 @@
+        Domain domain = domainHandler.verifyDomain(enterprise, template, descriptor);
+        Zone zone = zoneHandler.verifyZone(domain, descriptor);
+        Subnet subnet = subnetHandler.verifySubnet(zone, descriptor); // 2
+    }
+
+    private void startSession() {
+        System.out.println("Starting a new VSD session");
+        try {
+            session.start();
+        } catch (RestException e) {
+            e.printStackTrace();
+        }
+    }
+
+}
+{% endhighlight %}
+
+Run the main method from `NetworkBuilder` in your Java environment.  You should see the following results in your console :
+
+{% highlight console %}
+Starting a new VSD session
+Network requires Enterprise: MyLittleEnterprise
+Network requires Template: MyLittleDomainTemplate
+Network requires Domain: MyLittleDomain
+Network requires Zone: MyLittleWebTierZone
+Network requires Subnet: MyLittleWebSubnet1
+Enterprise MyLittleEnterprise already exists
+Template MyLittleDomainTemplate already exists
+Domain MyLittleDomain already exists
+Zone MyLittleWebTierZone already exists
+Subnet MyLittleWebSubnet1 not found.  Create one
+{% endhighlight %}
+
+If the code completed successfully, then in `VSA` you will see our complete network layout, with the Enterprise, Domain Template, Domain, Zone and Subnet objects now displayed.  From the left hand side of the main `VSA` screen, click the Enterprise object showing the name "MyLittleEnterprise" and then click the menu option 'Networks' at the top of the screen.  Ensure that the "Layer 3 Domains" item is selected on the left hand side of the Networks screen.  You may need to highlight the Zone icon to make the new Subnet appear.
+
+![VSA Enterprise and Subnet][VSA-Enterprise-And-Subnet]
+
+The `NetworkSubnetHandler` checks whether a VSD Subnet object exists before creating a new one.  You can test this by running `NetworkBuilder` again.  You should see the following results in your console :
+
+{% highlight console %}
+Starting a new VSD session
+Network requires Enterprise: MyLittleEnterprise
+Network requires Template: MyLittleDomainTemplate
+Network requires Domain: MyLittleDomain
+Network requires Zone: MyLittleWebTierZone
+Network requires Subnet: MyLittleWebSubnet1
+Enterprise MyLittleEnterprise already exists
+Template MyLittleDomainTemplate already exists
+Domain MyLittleDomain already exists
+Zone MyLittleWebTierZone already exists
+Subnet MyLittleWebSubnet1 already exists
+{% endhighlight %}
+
+# Removing VSD Objects
+In order to remove VSD objects, there are dependency rules that must be followed, based on parent-child relationships within the object tree.  For example, to remove a Zone object, there must not be any Subnet child objects attached.  Likewise, to remove a Domain object, you must first remove its Zone objects.  We also want to handle cases where we have more than one child object for a parent.  For simplicity, we will assume for this exercise that our network falls under a single common enterprise.

```

Comment:
Just a note that this is not exactly true. You can actually delete a whole domain without first deleting the subnet and zones, as long as there are no vPorts present.

However, for the sake of your blog post, no need to make a change here :)